pyquil/api/_compiler.py

```diff
@@ -190,177 +155,47 @@ def _collect_classical_memory_write_locations(
 
 
 class QPUCompiler(AbstractCompiler):
+    """
```

Comment:
The diff is pretty hard to interpret here, but overall, common code between `QPUCompiler` and `QVMCompiler` was moved up into the base class, `AbstractCompiler`.